sonic_platform_base/card_base.py

```diff
@@ -0,0 +1,80 @@
+#
+# card_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a card in SONiC chassis
+#
+
+class CardBase(object):
+    """
+    Abstract base class for implementing a platform specific class to
+    represent a control-card, line-card or fabric-card of a chassis
+    """
+
+    # Possible card types
+    MODULE_TYPE_CONTROL = "CONTROL-CARD"
+    MODULE_TYPE_LINE    = "LINE-CARD"
+    MODULE_TYPE_FABRIC  = "FABRIC-CARD"
+
+    def get_name(self):
+        """
+        Retrieves the name of the card
+
+        Returns:
+            string: The name of the card
+        """
+        raise NotImplementedError
+
+    def get_description(self):
+        """
+        Retrieves the description of the card
+
+        Returns:
+            string: The description of the card
+        """
+        raise NotImplementedError
+
+    def get_instance(self):
+        """
+        Retrieves the slot where the card is present
+
+        Returns:
+            string: slot representation, usually number
+        """
+        raise NotImplementedError
+
+    def get_type(self):
+        """
+        Retrieves the type of the card
+
+        Returns:
+            string: module-type MODULE_TYPE_CONTROL, MODULE_TYPE_LINE etc
+        """
+        raise NotImplementedError
+
+    def get_status(self):
```

Comment:
Introduce Empty, Offline, Present, Fault, Online states.

Reply:
what would be the default value?

Reply:
IMO, default to be Empty.
Platform to determine and update card's 'presence' state as Present.

Follow-on Q:
If platform detects card's presence later or in future, when LC is inserted in an empty slot later, platform would propagate the updated card state via get_change_event() API call?

Reply:
Follow up -
* Platform (software) driven fault management handling - bottom up approach.

Reply:
Both get_change_event() and fault_management handling are taken up as enhancement items.